lib/global.d.ts

```diff
@@ -49,9 +70,76 @@ declare global {
             tget<S = any>(
                 alias: `@${string}`,
                 options?: Partial<Loggable & Timeoutable & Withinable & Shadow>
-            ): Chainable<S | Subject>;
+            ): Chainable<S | Subject, NavigationShorthand>;
+
+            within(fn: WithinCallbackType): Chainable<Subject, NavigationShorthand>;
+            // inconsistent argument order
+            within(options: Partial<Loggable>, fn: WithinCallbackType): Chainable<Subject, NavigationShorthand>;
+
+            /**
+             * Access internal Cypress state. @see https://glebbahmutov.com/blog/cy-now-and-state/
+             * @param key If provided, equivalent to `cy.state()['key']`
+             */
+            state(): State;
+            state<StateKey extends keyof State>(key: StateKey): State[StateKey];
+        }
+
+        // @ts-expect-error No easy solution for this, but if we ignore the typing error, it works as expected.
```

Comment:
Figured out a solution using both an internal interface and a templated generic type wrapped around it, and will update the export before I merge this PR. Pseudo: `interface _ChainableLikeInternal<A,B> extends Chainable<A,B>{}` wrapped by `export type ChainableLike<A,B> = _ChainableLikeInner<A,B> & B`, so that Typescript doesn't complain about extending the generic type. The resulting IntelliSense mouseover wasn't indecipherable.